vllm/core/scheduler.py

```diff
@@ -1031,17 +1082,35 @@ def _can_append_slots(self, seq_group: SequenceGroup) -> bool:
             num_lookahead_slots=self._get_num_lookahead_slots(is_prefill),
         )
 
-    def schedule(self) -> Tuple[List[SequenceGroupMetadata], SchedulerOutputs]:
+    def _allow_output_proc_callback(self, seq_group: SequenceGroup) -> bool:
+        no_beam_search = (seq_group.sampling_params.n == 1
+                          and not seq_group.sampling_params.use_beam_search)
+
+        return no_beam_search
+
+    def schedule(
+        self
+    ) -> Tuple[List[SequenceGroupMetadata], SchedulerOutputs, List[Tuple[
+            ScheduledSequenceGroup, SequenceGroupMetadata]], bool]:
         # Schedule sequence groups.
         # This function call changes the internal states of the scheduler
         # such as self.running, self.swapped, and self.waiting.
         scheduler_start_time = time.perf_counter()
+
         scheduler_outputs = self._schedule()
         now = time.time()
 
         if not self.cache_config.enable_prefix_caching:
             common_computed_block_nums = []
 
+        # TODO: Combine multi-step and async postprocessor
+        allow_output_proc_callback: bool = (
```

Comment:
Why not just combining this logic into the _allow_output_proc_callback function?

Reply:
It may be possible, will see if I can do it.